client/app/components/account_config_input.coffee

```diff
@@ -0,0 +1,97 @@
+{div, label, input} = React.DOM
+{ErrorLine} = require './basic_components'
+classer = React.addons.classSet
+
+RouterMixin = require '../mixins/router_mixin'
+
+
+# Input used in the account configuration/creation form. An account input can
+# display an error message below it.
+module.exports = AccountInput = React.createClass
+    displayName: 'AccountInput'
+
+    mixins: [
+        RouterMixin
+        React.addons.LinkedStateMixin # two-way data binding
+    ]
+
+
+    getInitialState: ->
+        return @props
+
+
+    componentWillReceiveProps: (props) ->
+        @setState props
```

Comment:
There are no point in turning the props into state in this component since it only reads the props. You can simply use `@props.` rather than `@state.` and remove `getInitialState`  and `componentWillReceiveProps`. Putting **some** props into the state should be done carefully and is considered an anti-pattern (see https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html).